File: server/src/main/java/com/karelian/erl/Result.java
/*
 * @Author: Karelian_na
 */
package com.karelian.erl;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Result {

	/**
	 * 指示当前请求是否成功!
	 */
	private boolean success;

	/**
	 * 指示当前请求附带消息!
	 */
	private String msg;

	/**
	 * 指示当前请求返回的数据!
	 */
	private Object data;
	

	public static final Result PermissionNotAllowed = new Result("未经授权的操作!");

	public Result() {
		this.success = false;
		this.data = null;
		this.msg = "";
	}

	public Result(boolean success) {
		this.success = success;
		this.data = null;
		this.msg = null;
	}

	public Result(String msg) {
		this.success = false;
		this.data = null;
		this.msg = msg;
	}

	public Result(boolean success, Object data) {
		this.success = success;
		this.data = data;
		this.msg = null;
	}

	public Result(boolean success, String msg, Object data) {
		this.success = success;
		this.msg = msg;
		this.data = data;
	}
}

File: client/src/common/Result.ts
/**
 * @format
 * @Author: Karelian_na
 */

import type { AxiosResponse } from "axios";

export type Result = {
	code: number;
	success: boolean;
	msg: string;
	data: any;

	response?: AxiosResponse<Result, any>;
};

export class StatusCode {
	static readonly ERROR_EMPTY_REQUEST = 600001;
	static readonly ERROR_ILLEGAL_ACCESS = 600002;
	static readonly ERROR_PERMISSION_NOT_FOUND = 600003;
	static readonly ERROR_PERMISSION_NOT_ALLOWED = 600004;
	static readonly ERROR_UN_LOGIN = 600005;
	static readonly ERROR_UN_AUTHORIZED = 600006;
	static readonly ERROR_INVALID_ARGUMENT = 600007;
}

export type DailyPoemResult = {
	status: boolean;
	data: {
		id: string;
		content: string;
		popularity: number;
		origin: {
			title: string;
			dynasty: string;
			author: string;
			content: string[];
			translate: string[];
		};
		matchTags: string[];
		recommendedReason: string;
		cacheAt: string;
	};
	token: string;
	ipAddress: string;
};
